lib/vnet/setup.go

```diff
@@ -187,10 +187,16 @@ func AdminSubcommand(ctx context.Context, socketPath, ipv6Prefix, dnsAddr string
 	if err != nil {
 		return trace.Wrap(err, "getting TUN name")
 	}
-	if err := sendTUNNameAndFd(socketPath, tunName, tun.File().Fd()); err != nil {
+	if err := sendTUNNameAndFd(socketPath, tunName, tun.File()); err != nil {
 		return trace.Wrap(err, "sending TUN over socket")
```

Comment:
```suggestion
		_ = tun.Close()
		return trace.Wrap(err, "sending TUN over socket")
```

Reply:
(unless `tun` should stay open on error?)

Reply:
well the process is about to exit so it doesn't matter much tbh, but I wanted to defer the `tun.Close` instead of copying it 3 times, and decided to refactor a bit and simplify the goroutine/channel stuff here.

It was only written that way originally so the signatures of `createAndSetupTUNDeviceWith(out)?Root` could be the same, but we got rid of `createAndSetupTUNDeviceWithoutRoot` already, so now this doesn't have to be so convoluted